public/app/features/dashboard-scene/panel-edit/PanelEditorRenderer.tsx

```diff
@@ -82,6 +86,20 @@ function VizAndDataPane({ model }: SceneComponentProps<PanelEditor>) {
         <div {...primaryProps}>
           <vizManager.Component model={vizManager} />
         </div>
+        {showLibraryPanelSaveModal && libraryPanel instanceof LibraryVizPanel && (
+          <SaveLibraryVizPanelModal
+            libraryPanel={libraryPanel}
+            onDismiss={() => {
+              model.onDismissLibraryPanelModal();
+            }}
+            onConfirm={() => {
+              model.onConfirmSaveLibraryPanel();
+            }}
+            onDiscard={() => {
+              model.onDiscard();
+            }}
```

Comment:
Small one to avoid creating new functions per re-render
```suggestion
            onDismiss={model.onDismissLibraryPanelModal}
            onConfirm={model.onConfirmSaveLibraryPanel}
            onDiscard={model.onDiscard}
```

Reply:
Hmm, problem here is that it would bind `this` to the wrong object I think.

Reply:
If you make the model's methods arrow functions, it should be just fine @oscarkilhed